snippets/standard/datetime/json/datetime-converter-examples/example3/Program.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Buffers;
+using System.Buffers.Text;
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace DateTimeConverterExamples
+{
+    public class DateTimeConverterUsingDateTimeParseAsFallback : JsonConverter<DateTime>
+    {
+        public override DateTime Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
nit: Maybe add `Debug.Assert(typeToConvert == typeof(DateTime));` with a comment stating something similar to what @steveharter wrote: https://github.com/dotnet/samples/pull/1116#discussion_r309340116